src/dodal/devices/undulator_dcm.py

```diff
@@ -30,58 +38,83 @@ def _get_closest_gap_for_energy(
     return table[1][idx]
 
 
-class UndulatorDCM(Device):
+class UndulatorDCM(StandardReadable, Movable):
     """
-    Composite device to handle changing beamline energies
+    Composite device to handle changing beamline energies, wraps the Undulator and the
+    DCM. The DCM has a motor which controls the beam energy, when it moves, the
+    Undulator gap may also have to change to enable emission at the new energy.
+    The relationship between the two motor motor positions is provided via a lookup
+    table.
+
+    Calling unulator_dcm.set(energy) will move the DCM motor, perform a table lookup
+    and move the Undulator gap motor if needed. So the set method can be thought of as
+    a comprehensive way to set beam energy.
     """
 
-    class EnergySignal(Signal):
-        parent: "UndulatorDCM"
-
-        def set(self, value, *, timeout=None, settle_time=None, **kwargs) -> Status:
-            energy_kev = value
-            access_level = self.parent.undulator.gap_access.get(as_string=True)
-            if access_level == UndulatorGapAccess.DISABLED.value and not TEST_MODE:
-                raise AccessError(
-                    "Undulator gap access is disabled. Contact Control Room"
-                )
+    def __init__(self, prefix: str, undulator: Undulator, dcm: DCM, name: str = ""):
```

Comment:
`prefix` is unused and only there to satisfy `device_instantiation`, which assumes every device takes a prefix. I'm not overjoyed with this but unsure if this should be a change to `device_instantiation` or just not using to instantiate this device.

Reply:
given we plan to redo device instantiation, is it worth worrying about? IMO best to just note that we don't want this to be the case anymore when we do that